checker/src/abstract_value.rs

```diff
@@ -45,7 +46,10 @@ pub struct AbstractValue {
     pub expression_size: u64,
     /// Cached interval domain element computed on demand by get_as_interval.
     #[serde(skip)]
-    interval: RefCell<Option<Rc<IntervalDomain>>>,
+    pub interval: RefCell<Option<Rc<IntervalDomain>>>,
+    /// Cached tag domain element computed on demand by get_tags.
+    #[serde(skip)]
+    pub tags: RefCell<Option<Rc<TagDomain>>>,
```

Comment:
I think you want tags to be a map from tag type to tag value, where the latter is a lifted Boolean. (I.e. a domain whose elements are {}, {false}, {true}, {false, true} with BOTTOM={} (as always) and TOP={false, true}).